starknet_devnet/blueprints/base.py

```diff
@@ -126,25 +126,33 @@ def load():
 
 
 @base.route("/increase_time", methods=["POST"])
-def increase_time():
-    """Increases the block timestamp offset"""
+async def increase_time():
+    """Increases the block timestamp offset and generate block"""
     request_dict = request.json or {}
     time_s = extract_positive(request_dict, "time")
 
     state.starknet_wrapper.increase_block_time(time_s)
+    await state.starknet_wrapper.update_pending_state()
+    await state.starknet_wrapper.update_pending_block()
+    block = await state.starknet_wrapper.generate_latest_block()
 
-    return jsonify({"timestamp_increased_by": time_s})
+    return jsonify(
+        {"timestamp_increased_by": time_s, "block_hash": hex(block.block_hash)}
+    )
 
 
 @base.route("/set_time", methods=["POST"])
-def set_time():
-    """Sets the block timestamp offset"""
+async def set_time():
+    """Sets the block timestamp offset and generate block"""
     request_dict = request.json or {}
     time_s = extract_positive(request_dict, "time")
 
     state.starknet_wrapper.set_block_time(time_s)
+    await state.starknet_wrapper.update_pending_state()
+    await state.starknet_wrapper.update_pending_block()
```

Comment:
Does calling `generate_latest_block` take care of creating a pending block? Perhaps if there are pending transactions, we could first generate the new block, then set the new timestamp/offset and then create a new block for this timestamp to take effect. Would that simplify/complicate things?

Reply:
Ok, but do we want to have 2 new blocks in that case? I thought that we just want to have one block with increased time and pending transactions.

Reply:
I am aware that it would create two blocks. But if that would introduce any simplification or make us understand what we are doing and why we are calling which method, I'm in favor of that.

Reply:
discussed on the call and fixed in e8d3f075c7359d11fdc3910244a2a70e13e641eb